infor-design/enterprise
# 3096 - Fix a number of calendar bugs [v4.23.x]

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This PR address a number of calendar related issues on two issues. See the test notes for test cases

**Related github/jira issue (required)**:
Fixes #3096  
Fixed #3089

**Steps necessary to review your pull request (required)**:
**Test One**
- go to http://localhost:4000/components/calendar/example-index.html
- note that todays date is selected
- open the week view
- click today 
- the week should not change day

**Test two**
- go to http://localhost:4000/components/calendar/example-index.html
- click + and then add event with modal
- should look ok and all aligned
- switch theme to uplift
- still aligned, especially the checkbox, timepickers and icons on the datepicker where issues

**Test Three**
- go to http://localhost:4000/components/calendar/example-index.html
- click + and then add event with modal
- click the time picker to open it 
- click the x on the modal
- time picker will now shut with the modal

**Test Four**
- go to http://localhost:4000/components/timepicker/example-index.html
- open the time picker
- open the dropdowns
- the values should be what was initially selected when opening (before turned empty)

**Test Five**
- go to http://localhost:4000/components/calendar/example-index.html
- click a day or two in front of today
- click + and then add event with api -> do this repeatedly until a scroll bar appears
- the upcoming area should align fine and text not overlap
- try this test on IE Edge
